airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/parquet_parser.py

```diff
@@ -27,11 +29,16 @@ async def infer_schema(
         if not isinstance(parquet_format, ParquetFormat):
             raise ValueError(f"Expected ParquetFormat, got {parquet_format}")
 
-        # Pyarrow can detect the schema of a parquet file by reading only its metadata.
-        # https://github.com/apache/arrow/blob/main/python/pyarrow/_parquet.pyx#L1168-L1243
-        parquet_file = pq.ParquetFile(stream_reader.open_file(file, self.file_read_mode, logger))
-        parquet_schema = parquet_file.schema_arrow
+        with stream_reader.open_file(file, self.file_read_mode, logger) as fp:
+            parquet_file = pq.ParquetFile(fp)
+            parquet_schema = parquet_file.schema_arrow
+
+        # Inferred non-partition schema
         schema = {field.name: ParquetParser.parquet_type_to_schema_type(field.type, parquet_format) for field in parquet_schema}
+        # Inferred partition schema
+        partition_columns = {x.split("=")[0]: {"type": "string"} for x in self._extract_partitions(file.uri)}
```

Comment:
nit can we use a more descriptive variable name for the current partition than `x`